conda/__init__.py

```diff
@@ -148,3 +149,7 @@ def _default(self, obj):
 
 _default.default = JSONEncoder().default
 JSONEncoder.default = _default
+
+# Add s3:// support to urllib.parse
+urllib.parse.uses_netloc.append("s3")
+urllib.parse.uses_relative.append("s3")
```

Comment:
Can you elaborate why this is needed?

Reply:
@jezdez, are you talk about in the code or here?

This is need so that we can use `urllib.parse.urljoin` with s3. I will leave that in the code as well.

Reply:
I had to look this up as well by reading the Python source code. Changing global undocumented Python state is ugly.

Reply:
Yeah, this definitely isn't ideal 😭 (I felt gross writing it). Maybe there's a better way to handle url joins for s3 urls? 🤔 I'll take a look at this....

Reply:
@dholth, I've implemented an alternative approach. Please see the most recent commit.